iothub_client/samples/iothub_ll_c2d_sample/iothub_ll_c2d_sample.c

```diff
@@ -43,13 +43,53 @@
 #include "certs.h"
 #endif // SET_TRUSTED_CERT_IN_SAMPLES
 
+// Uncomment this define to use Asynchronous ACK of cloud-to-device messages.
```

Comment:
I think we should give a bit more guidance to user so they're not necessarily guessing.

How about extra lines underneath the "Uncomment..."
// Most applications should leave USE_C2D_ASYNC_ACK undefined.  The default sample behavior is easier for the application to manage and meets the vast majority of application scenarios.
// The pattern sampled when USE_C2D_ASYNC_ACK is defined should be used when the application does 
// not want to immediately acknowledge receipt of the message to Azure IoT Hub during the callback.  It wants
// instead to defer the acknowledgement until a later time with a call to IoTHubDeviceClient_LL_SendMessageDisposition.

Reply:
Oh - looks like i said something similiar in the "I really think we should add some additional lines" below, too.  I actually tihnk my write up above is better since it's more to point and it starts with common case of discouraging apps from using this more advanced feature which is what we want.

Reply:
Alright, I'll use this one then. Thanks John.